pennylane/qaoa/cycle.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Methods for finding max weighted cycle of weighted directed graphs
```

Comment:
```suggestion
Functionality for finding the maximum weighted cycle of directed graphs.
```
Currently this is visible in the docs, so just thought to polish.

Would be good to see if we can provide a reference for the problem.

Reply:
Hmm, for a max weighted cycle we could provide the [Johnson paper](https://www.cs.tufts.edu/comp/150GA/homeworks/hw1/Johnson%2075.PDF) but this only includes a classical approach...

Reply:
Right! Also does that paper talk about finding the max-weighted cycle, or just finding the set of cycles?